modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/SchemaIndexImpl.java

```diff
@@ -102,6 +102,9 @@ public SchemaIndexImpl(XSDSchema[] schemas) {
     
     public void destroy() {
         //remove the adapter from the schemas
+        if(schemas==null) {
```

Comment:
@ianturton formatting! There should be a space between {{if}} and the following bracket, and spaces around {{==}}.